legion/legion/k8s/properties.py

```diff
@@ -655,12 +671,15 @@ def _read_data_from_dict(self, source_dict):
 
         :param source_dict: dict with readed values
         :type source_dict: dict[str, str]
-        :return: None
+        :return: bool -- is new data has been received
         """
-        self._state = {
+        new_data = {
             k: base64.b64decode(v.encode('ascii')).decode('utf-8')
             for k, v in source_dict.items()
         }
+        data_has_been_changed = self._state != new_data
+        self._state = new_data
```

Comment:
@kirillmakhonin what is the reason to self._state = new_data  if  data_has_been_changed = false ?